### PR TITLE
musig2: fix early nonce gen option

### PR DESCRIPTION
**Pull request by sputn1ck**
Previously the early nonce generation option was not being respected when creating the context, with the `WithKnownSigners` option being used.

### CODE REVIEW

File: btcec/schnorr/musig2/context.go [hunk 1]
```diff
@@ -234,24 +234,24 @@ func NewContext(signingKey *btcec.PrivateKey, shouldSort bool,
...
-			if err != nil {
-				return nil, err
-			}
-		}
-
 	default:
 		return nil, ErrSignersNotSpecified
 	}
 
+	// If early nonce generation is specified, then we'll generate
```
Comment (Roasbeef) [inline btcec/schnorr/musig2/context.go:241]: Ah I guess my previous use case always had the _number_ of signers specified (case of LN channel funding).

**Review by Roasbeef — COMMENTED** [review]
Motivation makes sense, can we tack on a unit test to ensure the new code path works as expected and as a guard against regressions?

File: btcec/schnorr/musig2/context.go [hunk 2]
```diff
@@ -234,24 +234,24 @@ func NewContext(signingKey *btcec.PrivateKey, shouldSort bool,
...
+	// the nonce now to pass in to the session once all the callers
+	// are known.
```
Comment (guggero) [inline btcec/schnorr/musig2/context.go:243]: nit: we could re-flow the comment to only take up 2 lines now.

**Review by guggero — APPROVED** [review]
Nice fix, LGTM :tada:

**Review by Roasbeef — APPROVED** [review]
LGTM 🎎

**Pull request merged**